solidity/contracts/middleware/libs/InterchainAccountMessage.sol

```diff
@@ -4,187 +4,78 @@ pragma solidity >=0.8.0;
-            MessageType.CALLS,
-            _owner.addressToBytes32(),
-            _ism,
-            EMPTY_SALT
-        );
-        bytes memory suffix = abi.encode(_calls);
-        return bytes.concat(prefix, suffix);
-    }
-
-    /**
-     * @notice Returns formatted (packed) InterchainAccountMessage
-     * @dev This function should only be used in memory message construction.
-     * @param _owner The owner of the interchain account
-     * @param _ism The address of the remote ISM
-     * @param _calls The sequence of calls to make
-     * @return Formatted message body
-     */
-    function encode(
-        address _owner,
-        bytes32 _ism,
-        CallLib.Call[] calldata _calls
-    ) internal pure returns (bytes memory) {
-        return encode(TypeCasts.addressToBytes32(_owner), _ism, _calls);
+    function messageType(
+        bytes calldata _message
+    ) internal pure returns (MessageType) {
+        return MessageType(uint8(_message[0]));
     }
 
-    /**
-     * @notice Returns formatted (packed) InterchainAccountMessage
-     * @dev This function should only be used in memory message construction.
-     * @param _owner The owner of the interchain account
-     * @param _ism The address of the remote ISM
-     * @param _calls The sequence of calls to make
-     * @return Formatted message body
-     */
-    function encode(
-        bytes32 _owner,
-        bytes32 _ism,
-        CallLib.Call[] calldata _calls
-    ) internal pure returns (bytes memory) {
-        return encode(_owner, _ism, _calls, EMPTY_SALT);
+    function ism(bytes calldata _message) internal pure returns (bytes32) {
+        return bytes32(_message[33:65]);
     }
 
-    /**
-     * @notice Returns formatted (packed) InterchainAccountMessage
-     * @dev This function should only be used in memory message construction.
-     * @param _owner The owner of the interchain account
-     * @param _ism The address of the remote ISM
-     * @param _calls The sequence of calls to make
-     * @return Formatted message body
-     */
-    function encode(
-        address _owner,
-        bytes32 _ism,
-        CallLib.Call[] calldata _calls,
-        bytes32 _userSalt
-    ) internal pure returns (bytes memory) {
-        return
-            encode(TypeCasts.addressToBytes32(_owner), _ism, _calls, _userSalt);
+    function accountConfig(
+        bytes calldata _message
+    ) internal pure returns (AccountConfig memory) {
+        assert(messageType(_message) != MessageType.REVEAL);
+        return abi.decode(_message[1:97], (AccountConfig));
     }
+}
 
-    /**
-     * @notice Returns formatted (packed) InterchainAccountMessage
-     * @dev This function should only be used in memory message construction.
-     * @param _owner The owner of the interchain account
-     * @param _ism The address of the remote ISM
-     * @param _calls The sequence of calls to make
-     * @return Formatted message body
-     */
+library InterchainAccountMessageCalls {
```

Comment:
I still doubt that branching within your main logic to select a library is cleaner or more extensible than branching within your library code.

Reply:
idk about "cleaner" but minimizing the number of logical branches in safety critical code is imo always better